device/transport/amqp/test/_amqp_test.js

```diff
@@ -4,246 +4,1345 @@
+      disconnect: sinon.stub().callsArgWith(0, null, new results.Disconnected()),
+      initializeCBS: sinon.stub().callsArgWith(0, null),
+      putToken: sinon.stub().callsArgWith(2, null),
+      getReceiver: sinon.stub().callsArgWith(1, null, receiver),
+      attachSenderLink: sinon.stub().callsArgWith(2, null, sender),
+      attachReceiverLink: sinon.stub().callsArgWith(2, null, receiver),
+      detachSenderLink: sinon.stub().callsArg(0),
+      detachReceiverLink: sinon.stub().callsArg(0),
+      setDisconnectHandler: sinon.stub(),
+      send: sinon.stub().callsArgWith(3, null, new results.MessageEnqueued())
     };
+
+    transport = new Amqp(configWithSAS, fakeBaseClient);
   });
 
   afterEach(function () {
     transport = null;
     receiver = null;
+    sender = null;
+    fakeBaseClient = null;
   });
 
-  describe('#sendMethodResponse', function() {
-    /*Tests_SRS_NODE_DEVICE_AMQP_16_019: [The `sendMethodResponse` shall throw a `ReferenceError` if the `methodResponse` object is falsy.]*/
-    [null, undefined].forEach(function(badResponse) {
-      it('throws a ReferenceError if the methodResponse object is \'' + badResponse + '\'', function() {
-        assert.throws(function() {
-          transport.sendMethodResponse(badResponse, function() {});
-        }, ReferenceError);
+  describe('Direct Methods', function () {
+    describe('#sendMethodResponse', function () {
+      /*Tests_SRS_NODE_DEVICE_AMQP_16_019: [The `sendMethodResponse` shall throw a `ReferenceError` if the `methodResponse` object is falsy.]*/
+      [null, undefined].forEach(function (badResponse) {
+        it('throws a ReferenceError if the methodResponse object is \'' + badResponse + '\'', function () {
+          assert.throws(function () {
+            transport.sendMethodResponse(badResponse, function () {});
+          }, ReferenceError);
+        });
+      });
+
+      it('calls the callback with a NotConnectedError if the device is disconnected', function (testCallback) {
+        var fakeMethodResponse = { status: 200, payload: null, requestId: uuid.v4() };
+        transport.sendMethodResponse(fakeMethodResponse, function (err) {
+          assert.instanceOf(err, errors.NotConnectedError);
+          testCallback();
+        });
+      });
+
+      it('calls the callback with a NotConnectedError if the device is connecting', function (testCallback) {
+        var fakeMethodResponse = { status: 200, payload: null, requestId: uuid.v4() };
+        fakeBaseClient.connect = sinon.stub();
+        transport.connect();
+        transport.sendMethodResponse(fakeMethodResponse, function (err) {
+          assert.instanceOf(err, errors.NotConnectedError);
+          testCallback();
+        });
+      });
+
+      /*Tests_SRS_NODE_DEVICE_AMQP_16_020: [The `sendMethodResponse` response shall call the `AmqpDeviceMethodClient.sendMethodResponse` method with the arguments that were given to it.]*/
```

Comment:
>method with the arguments that were given to it [](start = 137, length = 47)

So what's the rational that this fails if disconnected but that onDeviceMethod will initiate a connect.  (At least that's how the requirments read).